includes/utils.php

```diff
@@ -483,6 +483,203 @@ function prepare_meta( $post_id ) {
 	return $prepared_meta;
 }
 
+/**
+ * Generates taxonomy term links for a given post.
+ *
+ * The code is taken from WP_Posts_List_Table::column_default and modified
+ * lightly to work in our context.
+ *
+ * @param string $taxonomy The taxonomy name.
+ * @param object $post     The post object.
+ * @param array  $terms    Optional. Array of terms.
+ *
+ * @return string The generated HTML for the taxonomy links.
+ */
+function generate_taxonomy_links( $taxonomy, $post, $terms = [] ) {
+	$taxonomy_object = get_taxonomy( $taxonomy );
+
+	if ( ! $taxonomy_object ) {
+		return '';
+	}
+
+	if ( ! $terms ) {
+		$terms = get_the_terms( $post, $taxonomy );
+	}
+
+	/**
+	 * Filter the taxonomy terms that should be synced.
+	 *
+	 * @since x.x.x
+	 * @hook dt_syncable_taxonomy_terms
+	 *
+	 * @param {array}  $terms    Array of terms.
+	 * @param {string} $taxonomy Taxonomy name.
+	 * @param {object} $post     Post Object.
+	 *
+	 * @return {array} Array of terms.
+	 */
+	$terms = apply_filters( "dt_syncable_{$taxonomy}_terms", $terms, $taxonomy, $post );
+
+	/**
+	 * Filter the terms that should be synced.
+	 *
+	 * @since x.x.x
+	 * @hook dt_syncable_terms
+	 *
+	 * @param {array}  $terms    Array of categories.
+	 * @param {string} $taxonomy Taxonomy name.
+	 * @param {object} $post     Post Object.
+	 *
+	 * @return {array} Array of categories.
+	 */
+	$terms = apply_filters( 'dt_syncable_terms', $terms, $taxonomy, $post );
+
+	if ( is_array( $terms ) ) {
+		$term_links = array();
+
+		foreach ( $terms as $t ) {
+			if ( is_array( $t ) ) {
+				$t = (object) $t;
+			}
+			$posts_in_term_qv = array();
+
+			if ( 'post' !== $post->post_type ) {
+				$posts_in_term_qv['post_type'] = $post->post_type;
+			}
+
+			if ( $taxonomy_object->query_var ) {
+				$posts_in_term_qv[ $taxonomy_object->query_var ] = $t->slug;
+			} else {
+				$posts_in_term_qv['taxonomy'] = $taxonomy;
+				$posts_in_term_qv['term']     = $t->slug;
+			}
+
+			$label = esc_html( sanitize_term_field( 'name', $t->name, $t->term_id, $taxonomy, 'display' ) );
+
+			$term_links[] = get_edit_link( $posts_in_term_qv, $label );
+		}
+
+		/**
+		 * Filters the links in `$taxonomy` column of edit.php.
+		 *
+		 * @since x.x.x
+		 * @hook dt_taxonomy_links
+		 *
+		 * @param string[]  $term_links Array of term editing links.
+		 * @param string    $taxonomy   Taxonomy name.
+		 * @param WP_Term[] $terms      Array of term objects appearing in the post row.
```

Comment:
Excuse me, I missed this one in the last review but we may need to follow the format we use in filter docs. Wrapping param types with `{}`. Also, we can add a doc comment for `@return`.

Reply:
Done